lib/internal/errors.js

```diff
@@ -681,15 +681,29 @@ E('ERR_INVALID_PROTOCOL',
   TypeError);
 E('ERR_INVALID_REPL_EVAL_CONFIG',
   'Cannot specify both "breakEvalOnSigint" and "eval" for REPL', TypeError);
+E('ERR_INVALID_RETURN_PROPERTY', (input, name, prop, value) => {
+  let type;
+  if (value && value.constructor && value.constructor.name) {
+    type = `instance of ${value.constructor.name}`;
+  } else {
+    type = `type ${typeof value}`;
+  }
+  return `Expected ${input} to be returned for the ${prop} from the ` +
+         `"${name}" function but got ${type}.`;
+}, TypeError);
+E('ERR_INVALID_RETURN_PROPERTY_STRING', (input, name, prop, value) => {
+  return `Expected a valid ${input} to be returned for the ${prop} from the ` +
+         `"${name}" function but got ${value}.`;
```

Comment:
Because this is quite a complex arguments case, I think it is important to have the labelled arguments and also the order is important to the message.